### DISCUSSION TITLE
Child Pages with dynamic number of childrens

### DISCUSSION THREAD

**Discussion by mndsl** [arkivanov/Decompose#462] [Q&A] [answered]
In one of our project, i need to display a horizontal pager with finite, but unknown at init time number of pages and configurations, because this information is taken from a remote server. How can this be achieved?

**Comment by arkivanov** [comment]
I think you can load your data from the server and then convert it to a list of configurations, and then call `navigaton.navigate { yourList }`. Make sure that every configuration in the list is unique by equality.

**Reply by mndsl** [reply]
This may work in this particular scenario, but the real question is in the title.
Consider a screen with (in)finite number of pages, which are fetched in portions of 100. 
When the user reaches near the end, a new set of 100 pages are loaded.

I think that one possible solution is to make initialPages to be a factory, which generates configurations on the fly and store them internally, instead of passing predefined list of configurations. But there must be a way to update the number of pages.

Is this possible with custom generic navigation?

**Reply by arkivanov** [reply, marked as answer]
You can have initialPages an empty list. Then load initial pages from the server and use `navigate { it.copy(items = it.items + <new_batch>) }` to append the batch. Then on the UI side, listen while the user scrolls, and at some point notify the component to load the next batch. Would this work?

**Reply by mndsl** [reply]
Yes, this is exactly what i needed, thank you!
